## Overview

Whenever a data professional works with a new dataset, the first step is to understand the context of the data during the discovering stage. Often, this involves discussing the data with project stakeholders and reading documentation about the dataset and the data collection process. After that, the data professional moves on to data cleaning and addresses issues like missing data, incorrect values, and irrelevant data. Computing descriptive stats is a common step to take after data cleaning.

In [1]:
import pandas as pd 
import numpy as np 
import datetime as dt 
import matplotlib.pyplot as plt

### Case of Study
I'm working for the government of a large nation, the government department of Education is seeking to understand the current literacy rates across the country.

##### Literacy Rate: 
The percentage of the population of a given age group that can read and write. 

### My Task:
Analyze data about the literacy rate among primary and secondary students. These are students who range in age from 6-18 years old.
I'll use descriptive stats to get a basic understanding of the literacy rate data for each district. 

In [3]:
edu_raw = pd.read_csv("./Raw_data/education_districtwise.csv")

edu_raw.head()

DISTNAME STATNAME  BLOCKS  VILLAGES  CLUSTERS  TOTPOPULAT  OVERALL_LI
0   DISTRICT32   STATE1      13       391       104    875564.0       66.92
1  DISTRICT649   STATE1      18       678       144   1015503.0       66.93
2  DISTRICT229   STATE1       8        94        65   1269751.0       71.21
3  DISTRICT259   STATE1      13       523       104    735753.0       57.98
4  DISTRICT486   STATE1       8       359        64    570060.0       65.00

In [4]:
edu_raw.shape

(680, 7)

In [5]:
edu_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 680 entries, 0 to 679
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   DISTNAME    680 non-null    object 
 1   STATNAME    680 non-null    object 
 2   BLOCKS      680 non-null    int64  
 3   VILLAGES    680 non-null    int64  
 4   CLUSTERS    680 non-null    int64  
 5   TOTPOPULAT  634 non-null    float64
 6   OVERALL_LI  634 non-null    float64
dtypes: float64(2), int64(3), object(2)
memory usage: 37.3+ KB


In [7]:
edu_raw.describe().round(2)

BLOCKS  VILLAGES  CLUSTERS   TOTPOPULAT  OVERALL_LI
count  680.00    680.00    680.00       634.00      634.00
mean    10.76    874.61    121.23   1899024.13       73.40
std      9.59    622.71     94.04   1547475.45       10.10
min      1.00      6.00      1.00      7948.00       37.22
25%      5.00    390.75     56.75    822694.00       66.44
50%      8.00    785.50    101.00   1564392.50       73.49
75%     13.00   1204.25    162.50   2587519.75       80.82
max     66.00   3963.00    592.00  11054131.00       98.76

In [9]:
edu_raw["OVERALL_LI"].describe().round(2)

count    634.00
mean      73.40
std       10.10
min       37.22
25%       66.44
50%       73.49
75%       80.82
max       98.76
Name: OVERALL_LI, dtype: float64

The dataset has 680 rows, after checking the describe funtion over the "OVERALL_LI" the count is 634, therefore there are some 
missing values for that column. 

In [10]:
edu_overall_li_nulls = edu_raw[ edu_raw["OVERALL_LI"].isnull()]
edu_overall_li_nulls

DISTNAME STATNAME  BLOCKS  VILLAGES  CLUSTERS  TOTPOPULAT  OVERALL_LI
54   DISTRICT302  STATE26       5       510        61         NaN         NaN
55   DISTRICT276  STATE26       6       393        59         NaN         NaN
200  DISTRICT588  STATE21      10       951        71         NaN         NaN
205  DISTRICT535  STATE21      13      1050        99         NaN         NaN
206  DISTRICT218  STATE21       8       341        47         NaN         NaN
207  DISTRICT258  STATE21       6       342        53         NaN         NaN
266  DISTRICT303   STATE3       4        62         5         NaN         NaN
267  DISTRICT608   STATE3       3       160        13         NaN         NaN
268   DISTRICT62   STATE3       4       145         7         NaN         NaN
269  DISTRICT474   STATE3       4        91        11         NaN         NaN
302  DISTRICT467  STATE14       9       159        40         NaN         NaN
303  DISTRICT449  STATE14       8        95        33         NaN         NaN
304  DISTRICT464  STATE14       9       162        46         NaN         NaN
305   DISTRICT38  STATE14       6        82        33         NaN         NaN
313  DISTRICT192   STATE4       2       503        36         NaN         NaN
314  DISTRICT164   STATE4       2       179        29         NaN         NaN
315  DISTRICT112   STATE4       2       297        38         NaN         NaN
316  DISTRICT426   STATE4       2       430        37         NaN         NaN
364  DISTRICT165  STATE34       7       776       107         NaN         NaN
389  DISTRICT378  STATE24      15      1187       157         NaN         NaN
390  DISTRICT591  STATE24       8       385        60         NaN         NaN
391  DISTRICT171  STATE24      14       996       149         NaN         NaN
435  DISTRICT376  STATE31       5       632       103         NaN         NaN
436  DISTRICT212  STATE31       2       244        30         NaN         NaN
437  DISTRICT439  STATE31       3       628        56         NaN         NaN
438  DISTRICT419  STATE31       4       512        61         NaN         NaN
439  DISTRICT445  STATE31       3       524        56         NaN         NaN
440  DISTRICT166  STATE31       5       771        93         NaN         NaN
441  DISTRICT317  STATE31       4       719        70         NaN         NaN
442   DISTRICT14  STATE31       6      1051       118         NaN         NaN
443  DISTRICT331  STATE31       5       786        85         NaN         NaN
444   DISTRICT59  STATE31       6       700       114         NaN         NaN
445  DISTRICT548  STATE31       6       688       111         NaN         NaN
496  DISTRICT233  STATE22       4       505        32         NaN         NaN
523  DISTRICT153   STATE9       6       629       161         NaN         NaN
524  DISTRICT179   STATE9       4       195        40         NaN         NaN
525  DISTRICT530   STATE9       4       269        83         NaN         NaN
526  DISTRICT139   STATE9       6       360        85         NaN         NaN
527  DISTRICT106   STATE9       6       648       135         NaN         NaN
528   DISTRICT89   STATE9       6       929       116         NaN         NaN
529  DISTRICT137   STATE9       5       402        82         NaN         NaN
568  DISTRICT506  STATE17       8      1006       151         NaN         NaN
601  DISTRICT105  STATE20       5       584        96         NaN         NaN
611  DISTRICT236  STATE20       8       687       164         NaN         NaN
612  DISTRICT383  STATE20       4       832       102         NaN         NaN
615  DISTRICT606  STATE20       6       782        96         NaN         NaN

In [11]:
edu_overall_li_nulls.shape

(46, 7)

#### function describre() for a Categorical Column

* Number of unique values
* Mode: most common value
* The frequency of the mode

In [14]:
edu_raw[["STATNAME","DISTNAME"]].describe()

STATNAME     DISTNAME
count       680          680
unique       36          680
top     STATE21  DISTRICT341
freq         75            1

In [18]:
range_overall_li = edu_raw["OVERALL_LI"].max() - edu_raw["OVERALL_LI"].min()
print(f"The Max level of Literacy is: {edu_raw["OVERALL_LI"].max()}")
print(f"The Minimun level of literacy is: {edu_raw["OVERALL_LI"].min()}")
print(f"The range of literacy across the country is: {range_overall_li.round(2)}")

The Max level of Literacy is: 98.76
The Minimun level of literacy is: 37.22
The range of literacy across the country is: 61.54
